### PR TITLE
Incorporate inflator into the TP queue

### PR DESCRIPTION
**Pull request by EdNoepel**
**Changes**
- Threshold prices are based upon a debt amount which changes over time.  To properly consider the time-value of debt, threshold prices are divided by the borrower's inflator snapshot when updating the queue, and multiplied by the inflator snapshot when reading the highest threshold price (HTP).
- Exposed the pending inflator, which was needed to calculate the TP offchain.
- Removed some unused cruft from the real-world test.

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
LGTM, I'd just use the shorthand if statement when calculating tp in order to reduce lines of code

**Comment by EdNoepel** [comment]
> LGTM, I'd just use the shorthand if statement when calculating tp in order to reduce lines of code

Done in d8b50d9ffb49bb8542e84798bdd0afa45d67bc01

**Review by mattcushman — COMMENTED** [review]
Should _threshold_price function be called somethiing else?  It's not really the current threshold price as I understand the code -- it would need to be multiplied by the current inflator value, right?  It's what the threshold price would have been at time t=0, maybe call it "_t0_extrapolated_threshold_price or something along those lines

File: src/ScaledPool.sol [hunk 1]
```diff
@@ -44,9 +44,9 @@ contract ScaledPool is Clone, FenwickTree, Queue {
     }
 
     struct Borrower {
-        uint256 debt;
-        uint256 collateral;
-        uint256 inflatorSnapshot;
+        uint256 debt;                // [WAD]
+        uint256 collateral;          // [WAD]
+        uint256 inflatorSnapshot;    // [WAD]
```
Comment (MikeHathaway) [inline src/ScaledPool.sol:49]: Are we still intending on updating `inflatorSnapshot` to `RAY`?

**Comment by EdNoepel** [comment]
> Should _threshold_price function be called somethiing else? It's not really the current threshold price as I understand the code -- it would need to be multiplied by the current inflator value, right? It's what the threshold price would have been at time t=0, maybe call it "_t0_extrapolated_threshold_price or something along those lines

Line 558 _does_ multiply it by the current inflator value.

File: src/ScaledPool.sol (hunk 1 above)
Comment (EdNoepel) [reply src/ScaledPool.sol:49]: Possibly.

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Comment by mattcushman** [comment]
> > Should _threshold_price function be called somethiing else? It's not really the current threshold price as I understand the code -- it would need to be multiplied by the current inflator value, right? It's what the threshold price would have been at time t=0, maybe call it "_t0_extrapolated_threshold_price or something along those lines
> 
> Line 558 _does_ multiply it by the current inflator value.

Line 558, as I read/understand it, multiplies the denominator by the inflator, is that correct?  Which is dividing by the inflator value.  So, it's really computing what the threshold price of this loan would have been, fully deflated, at time t==0.  The actual threshold price would be this times the current borrower inflator.

**Comment by EdNoepel** [comment]
> Line 558, as I read/understand it, multiplies the denominator by the inflator, is that correct? Which is dividing by the inflator value. So, it's really computing what the threshold price of this loan would have been, fully deflated, at time t==0. The actual threshold price would be this times the current borrower inflator.

Indeed; good catch.  Will resolve in my next PR.

**Pull request merged**